### PR TITLE
Fix [amp-list-load-more] doesn't work with [amp-state] bug

### PR DESCRIPTION
**Pull request by cathyxz**
Fixes https://github.com/ampproject/amphtml/issues/21195

Basically, the bug cause here is we're not correctly accounting for the fact that in `load-more` we append items and do not remove the original items in `amp-list` when we pass `removedElements` to `scanAndApply`. This code change fixes the bug, but I'm interested in understanding why we need to special case `amp-list`  in `amp-bind` with this `updateBindings` function and whether we need to continue doing so in the future.

### CODE REVIEW

**Comment by alabiaga** [comment]
lgtm. I am not clear on what you meant by the special case of handling bindings with this updateBindings function but @choumx would know. The updateBindings function has special handling of bindings based on attributes specific to the amp-list, does this answer your question?

File: build-system/routes/list.js [hunk 1]
```diff
@@ -74,6 +73,16 @@ const squareImgUrl = width => {
   return `http://picsum.photos/${width}?${randInt(50)}`;
 };
 
+const randomFalsy = () => {
+  const rand = Math.floor(Math.random() * Math.floor(3));
```
Comment (jridgewell) [inline build-system/routes/list.js]: `Math.floor(3)` is just `3`.

File: build-system/routes/list.js [hunk 2]
```diff
@@ -74,6 +73,16 @@ const squareImgUrl = width => {
...
+  switch (rand) {
+    case 1: return null;
+    case 2: return undefined;
+    case 3: return '';
```
Comment (jridgewell) [inline build-system/routes/list.js]: This case can never be hit.

File: extensions/amp-list/0.1/amp-list.js [hunk 3]
```diff
@@ -557,7 +557,7 @@ export class AmpList extends AMP.BaseElement {
     const isSSR = this.ssrTemplateHelper_.isSupported();
     let renderPromise = this.ssrTemplateHelper_.renderTemplate(
         this.element, current.data)
-        .then(result => this.updateBindings_(result))
+        .then(result => this.updateBindings_(result, current.append))
```
Comment (jridgewell) [inline extensions/amp-list/0.1/amp-list.js]: What type is `current` here?

Comment (cathyxz) [reply extensions/amp-list/0.1/amp-list.js]: It's defined as
```
    this.renderItems_ = {data, append, resolver, rejecter,
      payload: opt_payload};
```

And typed here: 
```
     * @private {?{data:(?JsonObject|string|undefined|!Array), resolver:!Function, rejecter:!Function}}
```

Is it a bit cleaner this way? 
```
/** @typedef {{
  data:(?JsonObject|string|undefined|!Array),
  resolver:!Function,
  rejecter:!Function,
  append:boolean,
}} */
export let RenderItems;
```

File: build-system/routes/list.js (hunk 2 above)
Comment (cathyxz) [reply build-system/routes/list.js]: Oops.

File: build-system/routes/list.js (hunk 1 above)
Comment (cathyxz) [reply build-system/routes/list.js]: Ohhhhhh I know what this is doing now. I floored the wrong variable lol.

File: extensions/amp-list/0.1/amp-list.js (hunk 3 above)
Comment (jridgewell) [reply extensions/amp-list/0.1/amp-list.js]: Thanks for looking, I was on my phone and couldn't find it.

The current type doesn't have an `append` (nor a `payload`). I'm not sure why this passes the type checker. We'll need to fix the type at least.

Comment (cathyxz) [reply extensions/amp-list/0.1/amp-list.js]: Yup. Done. Yeah this is kind of not-very-intelligent code written to make templating things other than list-items work. I wish we could refactor this, but it's a little bit high risk.

File: build-system/routes/list.js (hunk 1 above)
Comment (cathyxz) [reply build-system/routes/list.js]: Fixed this.

File: build-system/routes/list.js (hunk 2 above)
Comment (cathyxz) [reply build-system/routes/list.js]: Fixed this.

**Comment by cathyxz** [comment]
Merging this now so I can rebase a couple of dependent branches. Good call @jridgewell on the `renderItems_` section. Will try to take a look at this after adding some more tests.

**Comment by cathyxz** [comment]
Okay, this is a bit awkward since I realized too late that my type check fix didn't push. Will add that fix to a shortly upcoming PR.

**Pull request merged**